### ISSUE TITLE
why train and test take the same dataset.h5 ?

### ISSUE THREAD

**Issue by walidsliti** [nilmtk/nilmtk#666]
hello i have two  questions, first why the train and test take the same dataset that's not dividing 
![q1](https://user-images.githubusercontent.com/26322476/39242924-b8588f36-4883-11e8-8b94-746db22831c7.PNG)

Second here how we split the data at the 30th of april without giving time to the start and the end besides if we change the time to outbound time it still works why ?
![image](https://user-images.githubusercontent.com/26322476/39242997-f8ad0882-4883-11e8-81e4-46f7c8c12b2a.png)

**Comment by PMeira** [comment]
Not sure if I follow. The example trains in a subset of the data and then validates against the rest.

> without giving time to the start and the end

```python
>>> from pandas import Timestamp
>>> Timestamp('2000-01-01') == Timestamp('2000-01-01 00:00:00')
True
```

> still works why ?

Why should it not work?

**Comment by walidsliti** [comment]
@PMeira thanks for your help but 
in the first screen shot train and test took the same dataset redd.h5 so train = test they are the same dataset is it ? 

in the second screen shot i give time suppose to be out of bound because i use redd dataset which is in 2011 and 2012 besides i didn't get why  we put the start with test.window and end with train window 
i supposed that each line has its own start and end isn't it ?

**Comment by PMeira** [comment]
> thanks for your help but
> in the first screen shot train and test took the same dataset redd.h5 so train = test they are the same dataset is it ?

Not exactly, `set_window`  defines the subsets. Same source, different periods => different subsets.

I suggest you to plot some meters to understand what's happening. The `end` and `start` parameters are just constraints. For example

- If you provide only `end`, all data with timestamps up to `end` will be used. 
- If you provide only `start`, all data with timestamps after `start` will be used. It doesn't matter if there's no data until 2011.
